apps/web/src/pages/_app.tsx

```diff
@@ -165,4 +175,19 @@ const App = ({ Component, pageProps }: AppPropsWithLayout) => {
   )
 }
 
+MyApp.getInitialProps = async (context: AppContext) => {
```

Comment:
let's use cookies instead of getinitalprops. Getinitalprops will disable ssg

Reply:
do you think we should use cookies?. getInitial pros runs on server first time load, so it does run once on server and agin if u navigate to different path and back to base. if we use cookies like for example 
`
import { cookies } from 'next/headers'
 
export default function Page() {
  const cookiesList = cookies()
  const hasCookie = cookiesList.has('theme')
  return '...'
}

`
then were accessing this in the page. so would this not meant the result would take longer than getInital props since with getInitialProps its already defined before App?

Reply:
i was playing with cookies but similar issue https://github.com/vercel/next.js/issues/46356#issuecomment-1456680538

Reply:
compared to disable ssg, cookies has less impact on the overall performance

Reply:
ssg utilize edge distribution and pages with ssg won't hit the server at all

Reply:
I'm not getting the issue. If you can set the cookie to the response, then you're definitely be able to access the cookie on the client side

Reply:
read action on client side is not performance intense. The bottle neck of the performance should be on the server end